Problems linking gui application in debug mode

When I try to build the cutechess gui application in debug mode, linking fails due to

```
g++  -o cutechess .obj/boardscene.o .obj/boardview.o .obj/graphicsboard.o .obj/graphicspiece.o .obj/graphicspiecereserve.o .obj/piecechooser.o .obj/main.o .obj/chessclock.o .obj/engineconfigurationmodel.o .obj/engineconfigurationdlg.o .obj/mainwindow.o .obj/plaintextlog.o .obj/newgamedlg.o .obj/cutechessapp.o .obj/autoverticalscroller.o .obj/gamedatabasedlg.o .obj/pgnimporter.o .obj/gamedatabasemanager.o .obj/importprogressdlg.o .obj/pgndatabase.o .obj/pgngameentrymodel.o .obj/pgndatabasemodel.o .obj/engineoptiondelegate.o .obj/engineoptionmodel.o .obj/gamedatabasesearchdlg.o .obj/timecontroldlg.o .obj/engineconfigproxymodel.o .obj/gamewall.o .obj/tilelayout.o .obj/movelist.o .obj/pgntagsmodel.o .obj/newtournamentdialog.o .obj/engineselectiondlg.o .obj/gameviewer.o .obj/pathlineedit.o .obj/threadedtask.o .obj/stringvalidator.o .obj/pgntoken.o .obj/movenumbertoken.o .obj/movetoken.o .obj/movecommenttoken.o .obj/gametabbar.o .obj/evalhistory.o .obj/evalwidget.o .obj/settingsdlg.o .obj/enginemanagementwidget.o .obj/tournamentresultsdlg.o .obj/gamesettingswidget.o .obj/tournamentsettingswidget.o .obj/qcustomplot.o .obj/modeltest.o .obj/qrc_chessboard.o .obj/qrc_icons.o .obj/moc_boardscene.o .obj/moc_boardview.o .obj/moc_graphicspiece.o .obj/moc_piecechooser.o .obj/moc_chessclock.o .obj/moc_engineconfigurationmodel.o .obj/moc_engineconfigurationdlg.o .obj/moc_mainwindow.o .obj/moc_plaintextlog.o .obj/moc_newgamedlg.o .obj/moc_cutechessapp.o .obj/moc_autoverticalscroller.o .obj/moc_gamedatabasedlg.o .obj/moc_pgnimporter.o .obj/moc_gamedatabasemanager.o .obj/moc_importprogressdlg.o .obj/moc_pgndatabase.o .obj/moc_pgngameentrymodel.o .obj/moc_pgndatabasemodel.o .obj/moc_engineoptiondelegate.o .obj/moc_engineoptionmodel.o .obj/moc_gamedatabasesearchdlg.o .obj/moc_timecontroldlg.o .obj/moc_engineconfigproxymodel.o .obj/moc_gamewall.o .obj/moc_tilelayout.o .obj/moc_movelist.o .obj/moc_pgntagsmodel.o .obj/moc_newtournamentdialog.o .obj/moc_engineselectiondlg.o .obj/moc_gameviewer.o .obj/moc_pathlineedit.o .obj/moc_threadedtask.o .obj/moc_stringvalidator.o .obj/moc_gametabbar.o .obj/moc_evalhistory.o .obj/moc_evalwidget.o .obj/moc_settingsdlg.o .obj/moc_enginemanagementwidget.o .obj/moc_tournamentresultsdlg.o .obj/moc_gamesettingswidget.o .obj/moc_tournamentsettingswidget.o .obj/moc_qcustomplot.o .obj/moc_modeltest.o   -lcutechess -L/home/rikard/code/cutechess/projects/lib -lQt5Svg -lQt5PrintSupport -lQt5Widgets -lQt5Gui -lQt5Concurrent -lQt5Test -lQt5Core -lGL -lpthread 
.obj/modeltest.o: In function `ModelTest::data()':
/home/rikard/code/cutechess/projects/gui/3rdparty/modeltest/modeltest.cpp:444: undefined reference to `bool QTest::qCompare<int, unsigned int>(int const&, unsigned int const&, char const*, char const*, char const*, int)'
collect2: error: ld returned 1 exit status
```
The line it complains on is the following:
```
        int alignment = textAlignmentVariant.toInt();
        QCOMPARE( alignment, ( alignment & ( Qt::AlignHorizontal_Mask | Qt::AlignVertical_Mask ) ) );
```
where alignment is int, and (on my system) the Align*Mask is unsigned (I suppose it's an enum?) and QCOMPARE fails because they are different types.

According to the QT wiki (https://wiki.qt.io/Model_Test), there is a Qt5 repo here http://code.qt.io/cgit/qt/qtbase.git/tree/tests/auto/other/modeltest where this is changed to:
```
        Qt::Alignment alignment = textAlignmentVariant.value<Qt::Alignment>();
        QCOMPARE(alignment, (alignment & (Qt::AlignHorizontal_Mask | Qt::AlignVertical_Mask)));
```
Would you prefer this minimal change over updating the entire file (with some debugging disabled)? The diff would be fairly large, but it seems to be mostly cleanups (and a license change).

--- a/projects/gui/3rdparty/modeltest/modeltest.cpp
+++ b/projects/gui/3rdparty/modeltest/modeltest.cpp
@@ -440,7 +440,7 @@ void ModelTest::data()
     // Check that the alignment is one we know about
     QVariant textAlignmentVariant = model->data ( model->index ( 0, 0 ), Qt::TextAlignmentRole );
     if ( textAlignmentVariant.isValid() ) {
-        int alignment = textAlignmentVariant.toInt();
+        Qt::Alignment alignment = textAlignmentVariant.value<Qt::Alignment>();
         QCOMPARE( alignment, ( alignment & ( Qt::AlignHorizontal_Mask | Qt::AlignVertical_Mask ) ) );
     }
 

